### PR TITLE
Add Maestro tests for create & join rooms

### PR DESCRIPTION
**Pull request by julioromano**
I created 2 additional test users to use them as MXID we lookup during room creation.

I.e. we will not login as those users, it's just 2 known MXIDs that we know exist so we can look them up.

- https://github.com/vector-im/element-x-android/issues/409

### CODE REVIEW

File: .maestro/tests/roomList/createAndDeleteDM.yaml
Comment (julioromano) [reply .maestro/tests/roomList/createAndDeleteDM.yaml:8]: You're completely right though:
- we don't have these well known test users yet 
- the create/delete flows can easily break for the reasons outlined in https://matrix.to/#/!yMCNJKbESvGWRKcBOw:matrix.org/$b0Vr-Xn2dWWhe9qxo_nSP0ZD9afS5Cy8G-Hm3YMqj2A?via=matrix.org&via=element.io

**Comment by jmartinesp** [comment]
Do we want to merge this as is or are we waiting for a staging env?

**Comment by julioromano** [comment]
> Do we want to merge this as is or are we waiting for a staging env?

Coverting it back to draft as it's blocked by the "Empty room" bug which I'm investigate.

**Comment by julioromano** [comment]
Now it's ready for review an works correctly!

**Comment by jmartinesp** [comment]
Since we're out of quota for this month I tested the flow locally and it failed, when creating a room and searching for "maestroelement2" there are no results so the flow just stops:

```
Running on adb-9304164c-RWdDlv._adb-tls-connect._tcp.

 ║
 ║  > Flow
 ║
 ║    ✅  Run tests/init.yaml
 ║    ✅  Run tests/account/login.yaml
 ║    ✅  Run tests/settings/settings.yaml
 ║    ❌  Run tests/roomList/roomList.yaml
 ║      ✅  Run searchRoomList.yaml
 ║      ✅  Take screenshot build/maestro/300-RoomList
 ║      ✅  Run timeline/timeline.yaml
 ║      ✅  Run roomContextMenu.yaml
 ║      ❌  Run createAndDeleteRoom.yaml
 ║        ✅  Tap on "Create a new conversation or room"
 ║        ✅  Tap on "New room"
 ║        ✅  Tap on "Search for someone"
 ║        ✅  Input text maestroelement2
 ║        ✅  Tap on "maestroelement2", Index: 1
 ║        ❌  Tap on "Next"
 ║        🔲 Tap on "e.g. your project name"
 ║        🔲 Input text aRoomName
 ║        🔲 Tap on "What is this room about?"
 ║        🔲 Input text aRoomTopic
 ║        🔲 Tap on "Create"
 ║        🔲 Take screenshot build/maestro/320-createAndDeleteRoom
 ║        🔲 Tap on "aRoomName"
 ║        🔲 Tap on "Invite people"
 ║        🔲 Tap on "Search for someone"
 ║        🔲 Input text ${INVITEE2_MXID}
 ║        🔲 Tap on "${INVITEE2_MXID} ", Index: 1
 ║        🔲 Tap on "Send"
 ║        🔲 Tap on "Back"
 ║        🔲 Tap on "aRoomName"
 ║        🔲 Tap on "People"
 ║        🔲 Tap on "Back"
 ║        🔲 Tap on "Leave room"
 ║        🔲 Tap on "Leave"
 ║      🔲 Run createAndDeleteDM.yaml
 ║    🔲 Run tests/account/logout.yaml
 ║

Element not found: Text matching regex: Next
```

**Comment by julioromano** [comment]
> Since we're out of quota for this month I tested the flow locally and it failed, when creating a room and searching for "maestroelement2" there are no results so the flow just stops:
> 
> ```
> Running on adb-9304164c-RWdDlv._adb-tls-connect._tcp.
> 
>  ║
>  ║  > Flow
>  ║
>  ║    ✅  Run tests/init.yaml
>  ║    ✅  Run tests/account/login.yaml
>  ║    ✅  Run tests/settings/settings.yaml
>  ║    ❌  Run tests/roomList/roomList.yaml
>  ║      ✅  Run searchRoomList.yaml
>  ║      ✅  Take screenshot build/maestro/300-RoomList
>  ║      ✅  Run timeline/timeline.yaml
>  ║      ✅  Run roomContextMenu.yaml
>  ║      ❌  Run createAndDeleteRoom.yaml
>  ║        ✅  Tap on "Create a new conversation or room"
>  ║        ✅  Tap on "New room"
>  ║        ✅  Tap on "Search for someone"
>  ║        ✅  Input text maestroelement2
>  ║        ✅  Tap on "maestroelement2", Index: 1
>  ║        ❌  Tap on "Next"
>  ║        🔲 Tap on "e.g. your project name"
>  ║        🔲 Input text aRoomName
>  ║        🔲 Tap on "What is this room about?"
>  ║        🔲 Input text aRoomTopic
>  ║        🔲 Tap on "Create"
>  ║        🔲 Take screenshot build/maestro/320-createAndDeleteRoom
>  ║        🔲 Tap on "aRoomName"
>  ║        🔲 Tap on "Invite people"
>  ║        🔲 Tap on "Search for someone"
>  ║        🔲 Input text ${INVITEE2_MXID}
>  ║        🔲 Tap on "${INVITEE2_MXID} ", Index: 1
>  ║        🔲 Tap on "Send"
>  ║        🔲 Tap on "Back"
>  ║        🔲 Tap on "aRoomName"
>  ║        🔲 Tap on "People"
>  ║        🔲 Tap on "Back"
>  ║        🔲 Tap on "Leave room"
>  ║        🔲 Tap on "Leave"
>  ║      🔲 Run createAndDeleteDM.yaml
>  ║    🔲 Run tests/account/logout.yaml
>  ║
> 
> Element not found: Text matching regex: Next
> ```

Weird I've checked again and it does work on my machine.
Can you retry?

**Comment by julioromano** [comment]
@bmarty can you try out this tests too please?

**Comment by bmarty** [comment]
It does not work as well here, I have the same error:

image

Typing the full matrix id help, and confirm that the account does exists.

image

**Review by bmarty — APPROVED** [review]
I am approving the PR to see how maestro cloud will deal with the test suite. (and I have no remark on the code you've added 👍 )

Do not merge the PR yet please!

**Comment by jmartinesp** [comment]
> I am approving the PR to see how maestro cloud will deal with the test suite. (and I have no remark on the code you've added 👍 )

It seems like it failed, but this time at compile time, so it seems unrelated. Maybe it's using the code after merging with develop to build?

Also, for future tests instead of approving to run Maestro tests you can run a new GH action manually (Maestro actions have the `dispatch_workflow` entry point now).

**Comment by julioromano** [comment]
Rebased this to incorporate the latest changes to maestro test from @bmarty on `develop`

**Comment by julioromano** [comment]
FYI: https://github.com/vector-im/element-x-android/actions/runs/5452831665

**Comment by bmarty** [comment]
> Also, for future tests instead of approving to run Maestro tests you can run a new GH action manually (Maestro actions have the dispatch_workflow entry point now).

Good to know, thanks!

And testing it right now.

**Review by bmarty — APPROVED** [review]

**Comment by jmartinesp** [comment]
> https://github.com/vector-im/element-x-android/actions/runs/5452831665 did not report the status to this PR :/.

Yes, it doesn't associate the action with the PR so it won't publish any results here, sadly.

**Comment by julioromano** [comment]
New run: https://github.com/vector-im/element-x-android/actions/runs/5454009547

**Pull request merged**
